maven-resolver/src/main/java/org/wildfly/channel/maven/VersionResolverFactory.java

```diff
@@ -160,6 +169,37 @@ public List<File> resolveArtifacts(List<ArtifactCoordinate> coordinates) throws
                 throw new UnresolvedMavenArtifactException(ex.getLocalizedMessage(), ex, failed);
             }
         }
+
+        @Override
+        public String getReleaseVersion(String groupId, String artifactId) {
+            requireNonNull(groupId);
+            requireNonNull(artifactId);
+
+            final DefaultMetadata metadata = new DefaultMetadata(groupId, artifactId, "maven-metadata.xml", Metadata.Nature.RELEASE);
+            final List<MetadataRequest> requests = repositories.stream().map(r -> {
+                final MetadataRequest metadataRequest = new MetadataRequest();
+                metadataRequest.setMetadata(metadata);
+                metadataRequest.setRepository(r);
+                return metadataRequest;
+            }).collect(Collectors.toList());
+            final List<MetadataResult> metadataResults = system.resolveMetadata(session, requests);
+
+            final MetadataXpp3Reader reader = new MetadataXpp3Reader();
+            return metadataResults.stream()
+               .filter(r->r.getMetadata() != null)
+               .map(m->m.getMetadata().getFile())
+               .map(f-> {
+                   try {
+                       return reader.read(new FileReader(f));
+                   } catch (IOException | XmlPullParserException e) {
+                       throw new UnresolvedMavenArtifactException(e.getLocalizedMessage(), e, Set.of(new ArtifactCoordinate(groupId, artifactId, null, null, "*")));
+                   }
+               })
+               .map(m->m.getVersioning().getRelease())
```

Comment:
we can reuse this whole method and use `getVersioning().getLatest()` to let Maven determine the latest version.